src/Symfony/Component/Console/Command/CommandInterface.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Helper\HelperSet;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Base class for all commands.
+ */
+interface CommandInterface
```

Comment:
Doc above here is incorrect now. 

I also think this interface has far too many methods that are not relevant to implementing your command, they are merely a detail within Symfony. I would love to see this interface only define the methods you'd need to implement for a simple executable command. Perhaps multiple interfaces if you wish to implement additional features? But several of these methods are very specific to the architecture of the Command as implemented by Symfony.